src/enable_variable_ops/ngraph_rewrite_pass.cc

```diff
@@ -280,7 +280,11 @@ class NGraphEncapsulationPass : public NGraphRewritePass {
     }
 
     // 4. Encapsulate clusters then, if requested, dump the graphs.
-    TF_RETURN_IF_ERROR(EncapsulateClusters(options.graph->get(), idx));
+    FunctionDefLibrary* fdeflib_new = new FunctionDefLibrary();
+    TF_RETURN_IF_ERROR(
+        EncapsulateClusters(options.graph->get(), idx, fdeflib_new));
+    // TODO: not using fdeflib_new in this path. Only grappler path uses it
+    free(fdeflib_new);
```

Comment:
Do you need to free here, since as per your comment in ngraph_optimizer.cc - _According to the docs, the message assumes ownership and is responsible for clean up. a free on our part will probably trigger 2 frees._ ?

Reply:
To free or not to free, is the question, my friend!

Reply:
Freeing in case it is not used. Not freeing when handing it to `output` using `set_allocated_x` since that takes ownership in `ngraph_optimizer.cc`.